applications/processor/aggregator-processor/src/test/java/org/springframework/cloud/fn/aggregator/AggregatorProcessorTests.java

```diff
@@ -34,41 +39,60 @@
 public class AggregatorProcessorTests {
 
 	@Test
-	public void testWithJdbcMessageStore() {
+	public void testWithJdbcMessageStore() throws IOException {
 		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
 				TestChannelBinderConfiguration.getCompleteConfiguration(AggregatorProcessorTestApplication.class))
 				.web(WebApplicationType.NONE)
-				.run("--spring.cloud.function.definition=aggregatorFunction",
+				.run("--spring.cloud.function.definition=jsonBytesToMap|aggregatorFunction",
 						"--aggregator.message-store-type=jdbc")) {
 
 			InputDestination processorInput = context.getBean(InputDestination.class);
 			OutputDestination processorOutput = context.getBean(OutputDestination.class);
 
-			processorInput.send(
-					MessageBuilder.withPayload("2")
-							.setHeader(IntegrationMessageHeaderAccessor.CORRELATION_ID, "my_correlation")
-							.setHeader(IntegrationMessageHeaderAccessor.SEQUENCE_NUMBER, 2)
-							.setHeader(IntegrationMessageHeaderAccessor.SEQUENCE_SIZE, 2)
-							.build());
-			processorInput.send(
-					MessageBuilder.withPayload("1")
-							.setHeader(IntegrationMessageHeaderAccessor.CORRELATION_ID, "my_correlation")
-							.setHeader(IntegrationMessageHeaderAccessor.SEQUENCE_NUMBER, 1)
-							.setHeader(IntegrationMessageHeaderAccessor.SEQUENCE_SIZE, 2)
-							.build());
-
-			Message<byte[]> receive = processorOutput.receive(10_000, "aggregatorFunction-out-0");
-
-			assertThat(receive).isNotNull()
+			processorInput.send(createMessage("2", 2, 2));
+
+			processorInput.send(createMessage("1", 1, 2));
+
+			Message<byte[]> receive = processorOutput.receive(10_000, "jsonBytesToMapaggregatorFunction-out-0");
+
+			assertThat(receive)
 					.extracting(Message::getPayload)
 					.extracting(String::new)
 					.isEqualTo("[\"2\",\"1\"]");
+
+			ObjectMapper objectMapper = context.getBean(ObjectMapper.class);
+
+			Person person1 = new Person("First1 Last1", "St. #1");
+			processorInput.send(createMessage(objectMapper.writeValueAsBytes(person1), 2, 2));
+
+			Person person2 = new Person("First2 Last2", "St. #2");
+			processorInput.send(createMessage(objectMapper.writeValueAsBytes(person2), 1, 2));
+
+			receive = processorOutput.receive(10_000, "jsonBytesToMapaggregatorFunction-out-0");
+
+			assertThat(receive).isNotNull();
```

Comment:
Nice test addition! 

Wdyt about adding a check on that the `receive.payload` is instance of map? You do indirectly verify that via the `readValue` on 75.

Reply:
It is not a `Map`: the result of the `processorOutput.receive()` is a `byte[]`. That's why we have to deserialize it using an `ObjectMapper` and try to do that against expected types.

I cannot come up with an idea how to verify that input is a `Map` without a breaking use-case where it is really not a `Map`...

Reply:
Resolved.
See a fresh commit.